tee-worker/app-libs/parentchain-interface/src/integritee/event_handler.rs

```diff
@@ -378,10 +380,19 @@ where
 				.map_err(|_| ParentchainEventProcessingError::AssertionCreatedFailure)?;
 		}
 
+		if let Ok(events) = events.get_parentchain_block_proccessed_events() {
+			debug!("Handling ParentchainBlockProcessed events");
+			events.iter().for_each(|event| {
+				debug!("found ParentchainBlockProcessed event: {:?}", event);
+				// This is for monitoring purposes
+				handled_events.push(hash_of(ParentchainBlockProcessed::EVENT));
+			});
+		}
+
```

Comment:
Maybe a dump question: why not keep the same structure as the previous above block of `if let Ok(events)` ?

Reply:
In this case the function applied to each event is not fallible, that's why we use `for_each` and we don't need ownership of the event as don't use it (apart from the log), so we just use `iter()`. 

Is this what you mean by  "the same structure"?